spec/models/trip_spec.rb

```diff
@@ -39,4 +39,112 @@
       end
     end
   end
+  describe 'Analysis methods' do
+    context 'average duration' do
+      it 'returns the average duration of all rides' do
+        trips = create_list(:trip, 10)
+
+        average = trips.map(&:duration).sum/trips.count
+
+        expect(Trip.average_duration).to eq(average)
+      end
+      it 'returns the longest ride' do
+        trips = create_list(:trip, 10)
+
+        longest = trips.max_by(&:duration)
+
+        expect(Trip.longest_ride).to eq(longest)
+      end
+      it 'returns the shortest ride' do
+        trips = create_list(:trip, 10)
+
+        shortest = trips.min_by(&:duration)
+
+        expect(Trip.shortest_ride).to eq(shortest)
+      end
+      it 'returns the station with the most rides as a starting place' do
+        create_list(:station, 10)
+        trip1 = Trip.create(duration: 1000, start_date: Date.strptime('1/29/2013 9:08', '%m/%d/%Y %k:%M'), start_station_name: '2nd at South Park', start_station_id: 1, end_date: Date.strptime('8/29/2013 9:11', '%m/%d/%Y %k:%M'), end_station_name: '2nd at South Park', end_station_id: 100, bike_id: 1000, subscription_type: 'Subscriber', zip_code: '94703')
+        trip2 = Trip.create(duration: 1000, start_date: Date.strptime('2/27/2013 9:08', '%m/%d/%Y %k:%M'), start_station_name: '2nd at South Park', start_station_id: 2, end_date: Date.strptime('8/29/2013 9:11', '%m/%d/%Y %k:%M'), end_station_name: '2nd at South Park', end_station_id: 100, bike_id: 1000, subscription_type: 'Subscriber', zip_code: '94703')
+        trip3 = Trip.create(duration: 1000, start_date: Date.strptime('3/29/2013 9:08', '%m/%d/%Y %k:%M'), start_station_name: '2nd at South Park', start_station_id: 3, end_date: Date.strptime('8/29/2013 9:11', '%m/%d/%Y %k:%M'), end_station_name: '2nd at South Park', end_station_id: 100, bike_id: 1000, subscription_type: 'Subscriber', zip_code: '94703')
+        trip4 = Trip.create(duration: 1000, start_date: Date.strptime('4/29/2013 9:08', '%m/%d/%Y %k:%M'), start_station_name: '2nd at South Park', start_station_id: 4, end_date: Date.strptime('8/29/2013 9:11', '%m/%d/%Y %k:%M'), end_station_name: '2nd at South Park', end_station_id: 100, bike_id: 1000, subscription_type: 'Subscriber', zip_code: '94703')
+        trip5 = Trip.create(duration: 1000, start_date: Date.strptime('5/29/2013 9:08', '%m/%d/%Y %k:%M'), start_station_name: '2nd at South Park', start_station_id: 5, end_date: Date.strptime('8/29/2013 9:11', '%m/%d/%Y %k:%M'), end_station_name: '2nd at South Park', end_station_id: 100, bike_id: 1000, subscription_type: 'Subscriber', zip_code: '94703')
```

Comment:
I believe to solve this, you can write trips = create_list(:trip, 10, duration:1000, etc...)